exporter/awsemfexporter/metric_translator_test.go

```diff
@@ -792,7 +792,54 @@ func TestTranslateOtToCWMetricWithNameSpace(t *testing.T) {
 	assert.Equal(t, "myServiceNS", met.Measurements[0].Namespace)
 }
 
-func TestTranslateOtToCWMetricWithFiltering(t *testing.T) {
+func TestTranslateUnit(t *testing.T) {
+	metric := pdata.NewMetric()
+	metric.SetName("writeIfNotExist")
+
+	translator := &metricTranslator{
+		metricDescriptor: map[string]MetricDescriptor{
+			"writeIfNotExist": {
+				metricName: "writeIfNotExist",
+				unit:       "Count",
+				overwrite:  false,
+			},
+			"forceOverwrite": {
+				metricName: "forceOverwrite",
+				unit:       "Count",
+				overwrite:  true,
+			},
+		},
+	}
+
+	v := translator.translateUnit(&metric)
+	assert.Equal(t, "Count", v)
+
+	metric.SetUnit("ms")
```

Comment:
You should consider doing a table test here, which will make it far easier to spot which conditions are affected in case of test failures.

Reply:
Done